azpubsub/contextvalidator/src/test/java/AzPubSubPrincipalBuilderTest.java

```diff
@@ -0,0 +1,161 @@
+        MemberModifier.suppress(MemberMatcher.methodsDeclaredIn(Logger.class));
+
+        azPubSubPrincipalBuilder.configure(configs);
+    }
+
+    @Test
+    public void testConfigureSslContextValidatorClassNotExists() {
+        expectedEx.expect(IllegalArgumentException.class);
+        expectedEx.expectMessage("ClassNotFound");
+        expectedEx.expectMessage("mockNonExistingSslAuthenticationContextValidator");
+
+        Map<String, Object> configs = new java.util.HashMap<>();
+
+        configs.put("azpubsub.ssl.authentication.validator.class", "mockNonExistingSslAuthenticationContextValidator");
+        configs.put("azpubsub.sasl.authentication.validator.class", "mockNonExistingPositiveSaslAuthenticationContextValidator");
+        MemberModifier.suppress(MemberMatcher.methodsDeclaredIn(Logger.class));
+
+        azPubSubPrincipalBuilder.configure(configs);
+    }
+
+    @Test
+    public void testConfigureSaslContextValidatorClassNotExists() {
+        expectedEx.expect(IllegalArgumentException.class);
+        expectedEx.expectMessage("ClassNotFound");
+        expectedEx.expectMessage("mockNonExistingPositiveSaslAuthenticationContextValidator");
+
+        Map<String, Object> configs = new java.util.HashMap<>();
+
+        configs.put("azpubsub.ssl.authentication.validator.class", "mockPositiveSslAuthenticationContextValidator");
+        configs.put("azpubsub.sasl.authentication.validator.class", "mockNonExistingPositiveSaslAuthenticationContextValidator");
+        MemberModifier.suppress(MemberMatcher.methodsDeclaredIn(Logger.class));
+
+        azPubSubPrincipalBuilder.configure(configs);
+    }
+
+    @Test
+    public void testValidateSslAuthenticationContextPositive() {
+        Map<String, Object> configs = new java.util.HashMap<>();
+
+        configs.put("azpubsub.ssl.authentication.validator.class", "mockPositiveSslAuthenticationContextValidator");
+        configs.put("azpubsub.sasl.authentication.validator.class", "mockPositiveSaslAuthenticationContextValidator");
+        MemberModifier.suppress(MemberMatcher.methodsDeclaredIn(Logger.class));
+
+        SSLSession sslSession = EasyMock.mock(SSLSession.class);
+        InetAddress inetAddress =  InetAddress.getLoopbackAddress();
+
+        azPubSubPrincipalBuilder.configure(configs);
+
+        SslAuthenticationContext sslAuthenticationContext = new SslAuthenticationContext(sslSession, inetAddress);
+        KafkaPrincipal kafkaPrincipal = azPubSubPrincipalBuilder.build(sslAuthenticationContext);
+        assert(kafkaPrincipal.getName().equals("ANONYMOUS"));
+        assert(KafkaPrincipal.ANONYMOUS.getPrincipalType().equals(kafkaPrincipal.getPrincipalType()));
+        assert(KafkaPrincipal.ANONYMOUS.getName().equals(kafkaPrincipal.getName()));
+    }
+
+    @Test
+    public void testValidateSaslAuthenticationContextPositive() {
+        Map<String, Object> configs = new java.util.HashMap<>();
```

Comment:
I think this is where you need @BeforeClass set up, The Map and configs are the same in multiple tests.

Reply:
Configs are the same in multiple tests, but it doesn't mean that all the tests have the exactly same settings.  Different configs is also a valid test case.